source/extensions/filters/http/cache/simple_http_cache/simple_http_cache.cc

```diff
@@ -148,19 +148,25 @@ void SimpleHttpCache::updateHeaders(const LookupContext& lookup_context,
   const auto& simple_lookup_context = static_cast<const SimpleLookupContext&>(lookup_context);
   const Key& key = simple_lookup_context.request().key();
   absl::WriterMutexLock lock(&mutex_);
-
   auto iter = map_.find(key);
   if (iter == map_.end() || !iter->second.response_headers_) {
     on_complete(false);
     return;
   }
-  auto& entry = iter->second;
-
-  // TODO(tangsaidi) handle Vary header updates properly
-  if (VaryHeaderUtils::hasVary(*(entry.response_headers_))) {
-    on_complete(false);
-    return;
+  if (VaryHeaderUtils::hasVary(*iter->second.response_headers_)) {
+    auto varied_key =
```

Comment:
please use an explicit type here as the type returned by variedRequestKey is not obvious at the call-site.